src/evict.c

```diff
@@ -485,6 +486,30 @@ static unsigned long evictionTimeLimitUs() {
     return ULONG_MAX;   /* No limit to eviction time */
 }
 
+struct EvictDB{
+    int dbid;
+    dict* sampleDict;
+    dict* keyDict;
+};
+
+static int getDbsToEvict(struct EvictDB * dbs) {
+    int i, j=0;
+    i = (uint32_t)random()%server.dbnum;
```

Comment:
I'm not sure why you're doing random here?

Reply:
We suppport server.maxmemory_samples smaller than dbnum.